### PR TITLE
Interaction Fingerprint Addition

### PR DESCRIPTION
**Pull request by rbharath**
This PR adds in a series of interaction fingerprints that compute interactions between proteins and ligands. These are pulled in from the old refactor PR #1792.

### CODE REVIEW

**Comment by rbharath** [comment]
Ok, I think this is ready for first round of review!

@ncfrey Would you mind doing a first review pass? It looks like you'll be the first user of the fingerprints so would love your feedback on them

CC @nd-02110114 @peastman who might also be interested

File: deepchem/utils/rdkit_utils.py [hunk 1]
```diff
@@ -370,3 +376,244 @@ def merge_molecules(molecules):
...
+  ring_xyz = np.zeros((len(ring_indices), 3))
+  for i, atom_idx in enumerate(ring_indices):
+    atom_position = conformer.GetAtomPosition(atom_idx)
+    ring_xyz[i] = np.array(atom_position)
+  ring_centroid = compute_centroid(ring_xyz)
+  return ring_centroid
+
+
+def get_contact_atom_indices(fragments: List, cutoff: float = 4.5) -> List:
+  """Compute that atoms close to contact region.
```
Comment (ncfrey) [inline deepchem/utils/rdkit_utils.py:506]: "the atoms"

File: deepchem/utils/rdkit_utils.py [hunk 2]
```diff
@@ -370,3 +376,244 @@ def merge_molecules(molecules):
...
+  cutoff: float
+    The cutoff distance in angstroms.
+
+  Returns
+  -------
+  A list of length `len(molecular_complex)`. Each entry in this list
+  is a list of atom indices from that molecule which should be kept, in
+  sorted order.
+  """
+  # indices to atoms to keep
```
Comment (ncfrey) [inline deepchem/utils/rdkit_utils.py:531]: "of atoms"

File: deepchem/utils/rdkit_utils.py [hunk 3]
```diff
@@ -370,3 +376,244 @@ def merge_molecules(molecules):
...
+
+  # Now extract atoms
+  #atoms_to_keep = []
+  #for i, frag_keep_inds in enumerate(keep_inds):
+  #  frag = fragments[i]
+  #  mol = frag[1]
+  #  atoms = mol.GetAtoms()
+  #  frag_keep = [atoms[keep_ind] for keep_ind in frag_keep_inds]
+  #  atoms_to_keep.append(frag_keep)
+  #return atoms_to_keep
```
Comment (ncfrey) [inline deepchem/utils/rdkit_utils.py:556]: Delete?

**Review by ncfrey — COMMENTED** [review]
I took a first pass through, awesome new featurizers! There's a lot here. I think two main things that I thought of were: it would be nice to consolidate the vectorization/voxelization loop into a single function or two functions to reduce code duplication if possible; and because I'm not too familiar with interaction fingerprints, it would be helpful to add some text in the `ComplexFeaturizer` docs to explain why you might want to use certain featurizers.

**Comment by gabegrand** [comment]
Just wanted to say I think this is an awesome addition to DeepChem. I'm particularly interested in the `ContactCircularFingerprint` and `ContactCircularVoxelizer` featurizers. Looking forward to trying these out once this PR is merged!

**Comment by rbharath** [comment]
I've addressed @ncfrey's comments from above. There's still a few issues that aren't quite 100%, but I think this PR is good enough to merge in to unblock @ncfrey's tutorial work. I'll plan to merge in once Travis passes

**Comment by rbharath** [comment]
Hmm, there's some tricky `mypy` errors that have to do with type signatures. I think I need to generalize a couple of type signatures to be more general which is taking a bit of thought. Will continue iterating...

**Comment by ncfrey** [comment]
From a quick look-through, it looks like some of the functions are expecting more explicit types than are returned by some other utilities. If possible, it would be nice for `rdkit_utils.load_complex` to return `List[Tuple[np.ndarray, RDKitMol]]`, if that is the preferred type signature for fragments. 

A lot of the other errors look related to the `hash_function` arg of `voxel_utils.voxelize`. I think this can be `hash_function: Optional[Callable[..., Any]] = None`, because if `hash_function` isn't specified the feature tensor will just be `0` along the fourth dimension.

**Comment by ncfrey** [comment]
I think all the errors related to this PR are now fixed, but there are three unrelated test failures:
```
FAILED deepchem/feat/tests/test_graph_data.py::TestGraph::test_graph_data - I...
FAILED deepchem/feat/tests/test_smiles_tokenizer.py::TestSmilesTokenizer::test_tokenize
FAILED deepchem/models/tests/test_attentivefp.py::test_attentivefp_classification
```

**Comment by rbharath** [comment]
@ncfrey I think these test failures are unrelated and @seyonechithrananda is working on a fix, so I think we can go ahead and merge in this PR to unblock your work. Thanks for the test fixes!

**Pull request merged**